File: ios/Classes/CacheUtil.swift
//
//  CacheUtil.swift
//  ivs_player
//
//  Created by Utkarsh Sharma on 09/03/23.
//

import Foundation
import AmazonIVSPlayer


class CacheUtil {
    //Singleton Class for maintaining cache at native side.
    static let i = CacheUtil()
    private init(){}
    
     private var maxInstanceCount: Int = 5
     private var playerCache: [Int32:IvsPlayer]=[:]
    
     func getPlayer(key:Int32) -> IvsPlayer? {
        return playerCache[key]
    }
    
     func setPlayer(key:Int32,playerInstance:IvsPlayer) {
        playerCache[key] = playerInstance
    }
    
    func removePlayer(key:Int32){
        print(getPlayer(key: key))
        playerCache.removeValue(forKey: key)
        print(getPlayer(key: key))
    }
}

File: android/src/main/kotlin/com/example/ivs_player/CacheUtil.kt
package com.example.ivs_player

import android.content.Context
import com.amazonaws.ivs.player.PlayerView
import io.flutter.view.TextureRegistry
import io.flutter.view.TextureRegistry.SurfaceTextureEntry


object CacheUtil {
    //Singleton Class for maintaining cache at native side.
//    static let i = CacheUtil()
//    private init(){}
//
    private var maxInstanceCount: Int = 5
    private var playerCache = HashMap<Long,IvsPlayer >()
//
    fun getPlayer(key:Long): IvsPlayer? {
        return playerCache[key]
    }


    fun setPlayer(key:Long,playerInstance:IvsPlayer) {
        playerCache[key] = playerInstance
    }

    fun removePlayer(key:Long){
//        playerCache[key]?.player?.removeListener(ivsPlayerView)
        playerCache.remove(key)
    }
}

class CacheModel{
    lateinit var ivsPlayer: IvsPlayer
    lateinit var surfaceTextureEntry: TextureRegistry.SurfaceTextureEntry
}